spec/models/piece_spec.rb

```diff
@@ -81,15 +105,54 @@
       expect(piece1.obstructed?(100, 300)).to eq true
       # it raises an error because there is an invalid input. There is no row: 100, col: 300
     end
+  end
 
-    it 'returns true for invalid destination' do
+  describe 'move_to!' do
+    it 'moves the piece if the spot is empty' do
+      game = FactoryGirl.create(:game)
+      piece = game.pieces.find_by_current_row_index_and_current_column_index(1, 0)
+      piece.move_to!(2, 0)
+      expect(piece.current_row_index).to eq 2
+      expect(piece.current_column_index).to eq 0
+    end
+
+    it 'returns true for spot taken' do
       game = FactoryGirl.create(:game) # Would do this using FactoryGirl
-      piece1 = FactoryGirl.create(:piece, game: game, current_row_index: 1, current_column_index: 1)
-      FactoryGirl.create(:piece, game: game, current_row_index: 2, current_column_index: 2)
+      piece = game.pieces.find_by_current_row_index_and_current_column_index(0, 0)
 
-      expect(piece1.obstructed?(2, 2)).to eq true
-      # expect(response).to have_http_status(:unproceessable_entity)
+      expect(piece.spot_taken?(1, 1)).to eq true
       # it raises an error because there is a piece in the destination.
     end
```

Comment:
For this test, since you're testing the `spot_taken?` method. You'd want to wrap it in it's own describe block

```Ruby
describe 'spot_taken?' do
  etc etc
end
```